scipy/optimize/_minpack_py.py

```diff
@@ -584,8 +584,9 @@ def curve_fit(f, xdata, ydata, p0=None, sigma=None, absolute_sigma=False,
     xdata : array_like
         The independent variable where the data is measured.
         Should usually be an M-length sequence or an (k,M)-shaped array for
-        functions with k predictors, and each element should be float
-        convertible if it is an array like object.
+        functions with k predictors, and a (k,M,N)-shaped 3D array is also 
+        acceptable. Each element should be float convertible if it is an 
+        array like object.
```

Comment:
When I suggested documenting 3D arrays, I had in mind that we should
a) make a more explicit decision about whether to support this officially and 
b) if so, document precisely what it means. Otherwise, it is not clear how it behaves. 

So let's take care of that in a separate PR. 
```suggestion
        functions with k predictors, and each element should be float
        convertible if it is an array like object.
```
Do you have a preference about whether it should be supported?

Reply:
My preference  is supporting N-d array if possible. Because, in some fields which we are not familiar with, higher degree predictor might be useful. But I agree we don’t specify it is 3D here now.